### ISSUE TITLE
Why does marathon need to resolve the hostname of the node it runs on when --hostname is set?

### ISSUE THREAD

**Issue by F21** [mesosphere/marathon#2153]
I am running marathon 0.10.1 using this command:

```
 /opt/test/marathon/bin/start --master zk://192.168.33.10:2181/mesos --zk zk://192.168.33.10:2181/marathon --hostname 192.168.33.10
```

Unfortunately, marathon fails with the following error:

```
Failed to obtain the IP address for 'core-01'; the DNS service may not be able to resolve it: Name or service not known
```

`core-01` is the hostname of the machine marathon is running on. I can get it to start properly if I add the following entry into the `/etc/hosts` file:

```
192.168.33.10 core-01
```

However, I would like to reduce the administrative burden and not have to make changes to the `/etc/hosts` file. The best way to do this 

The error occurs even if I set the `--hostname` flag. Why is marathon trying to resolve the hostname of my machine even though I have set the `--hostname` flag?

**Comment by pierlo-upitup** [comment]
Hi @F21, thanks for reporting this. 

In your pasted command I see `--hostname 192.168.33.10` –  I suppose you meant to write `--hostname core-01`?

As per the [CLI flags docs page](https://mesosphere.github.io/marathon/docs/command-line-flags.html) : 

> `--hostname` (Optional. **Default: hostname of machine**): The advertised hostname stored in ZooKeeper so another standby host can redirect to the elected leader. Note: Default is determined by InetAddress.getLocalHost.

`hostname` is by default the hostname of the machine. Could you help me understand better what you're trying to achieve?

**Comment by F21** [comment]
@pierlo-upitup 

Thanks for getting back to me! I am running mesos + marathon on CoreOS natively (not insider docker containers).

I am currently using vagrant, so the hostname of the machine does not have an entry inside `/etc/hosts`.

I also prefer to use ip addresses over hostnames, so that when I connect to the cluster, I don't need to edit the `/etc/hosts` files on my host to manage the services and machines running inside the cluster.

When starting mesos, I am able to do this to avoid using hostnames:

`./mesos-master --ip 192.168.33.10 --hostname 192.168.33.10`

Does a similar method exist for marathon?

**Comment by pierlo-upitup** [comment]
Thanks for clarifying the scenario. 

cc @meichstedt @aquamatthias

**Comment by amouat** [comment]
I'm facing the same problem, and this used to work for me. I note the marathon 0.9.1 docker image was updated 5 days ago - did this break something?

**Comment by cmaloney** [comment]
Could we also get a check inside marathon that the hostname it is announcing for leader election is at least locally resolvable? Inside DCOS we had a case for a little bit where some bash escaping was wrong, so rather than a actual hostname we got a great big shell-expression, both this particular issue and that would be checked by doing a sanity check "is this resolvable".

Otherwise issues will only show up on non-leading masters when a master has failed, which is a bad time to run into other unexpected failures.

**Comment by anguslees** [comment]
cmaloney: That sounds like a fine idea, but we're already being outsmarted by a "clever" hostname verification check in marathon.  I'd sure like to have "do-what-I-say" working first :/

**Comment by anguslees** [comment]
Turns out this code is actually in libprocess: https://github.com/apache/mesos/blob/master/3rdparty/libprocess/src/process.cpp#L939

A workaround is to set `LIBPROCESS_IP=$ip`

I'm curious why I haven't also hit this with mesos-{master,slave}.  Perhaps they (or one of the wrapper scripts I'm using) sets LIBPROCESS_IP...

**Comment by kolloch** [comment]
This should be fixed with #1671. Correct? Otherwise please reopen.
